src/Auth0.php

```diff
@@ -334,18 +347,40 @@ public function getBearerToken(
 
     public function getCredentials(): ?object
     {
-        $user = $this->getState()->getUser();
+        $state = $this->getState();
+        $user = $state->getUser();
 
         if (null === $user) {
             return null;
         }
 
-        $idToken = $this->getState()->getIdToken();
-        $accessToken = $this->getState()->getAccessToken();
-        $accessTokenScope = $this->getState()->getAccessTokenScope();
-        $accessTokenExpiration = (int) $this->getState()->getAccessTokenExpiration();
+        $idToken = $state->getIdToken();
+        $accessToken = $state->getAccessToken();
+        $accessTokenScope = $state->getAccessTokenScope();
+        $accessTokenExpiration = (int) $state->getAccessTokenExpiration();
         $accessTokenExpired = time() >= $accessTokenExpiration;
-        $refreshToken = $this->getState()->getRefreshToken();
+        $refreshToken = $state->getRefreshToken();
+        $backchannel = $state->getBackchannel();
+
+        // If this is an authenticated pseudo-stateful session (i.e. not authorizing an access token) ...
+        if (null !== $idToken) {
+            $cache = $this->configuration()->getBackchannelLogoutCache();
+
+            // Does the session have a backchannel key available for lookup?
+            $backchannel = $state->getBackchannel();
```

Comment:
I'm not sure if this is expensive or not? If it is, it should be behind the backchannel enabled config (which I assume is `$cache instanceof \Psr\Cache\CacheItemPoolInterface`)

Reply:
Appreciate the caution, but should be all good:

- `$cache` will be `null` if Backchannel is not configured for the SDK, and thus ignored. Otherwise, the cache will have been initialized along with the SDK at runtime by necessity. It only occurs max once per request, and I imagine in 99% of cases will be the same cache resource the developer is already using for other things, like JWKS caching, so overhead will likely be non-existent in most applications.

- `setBackchannel()` just reads the string UUID stored in the device cookie's JSON structure, which is already extracted and decoded at request time.